### DISCUSSION TITLE
How to fix "Type instantiation is excessively deep and possibly infinite" Error

### DISCUSSION THREAD

**Discussion by p3yman** [ardatan/feTS#379] [Q&A] [answered]
I'm exploring this amazing tool but I'm stuck in a weird error.
 The project is a simple Vite + React to test using Spotify API. 

### What I did:
1- Created a `openapi.ts` file that includes `export const openapi = {} as const`;

2- Created the `client.ts` file:
```ts
import { createClient, NormalizeOAS } from "fets";
import type { openapi } from "./openapi";

export const client = createClient<NormalizeOAS<typeof openapi>>({
  endpoint: "https://api.spotify.com/v1",
});
```
3- Used the client this to the `App.tsx`:
```ts
import { NormalizeOAS, OASOutput } from "fets";
import type { openapi } from "./api/openapi";
import { useEffect, useState } from "react";
import { client } from "./api/client";

type Artist = OASOutput<
  NormalizeOAS<typeof openapi>,
  "/artists/{id}",
  "get",
  "200"
>;

function App() {
  const [artist, setArtist] = useState<Artist>();

  useEffect(() => {
    client["/artists/{id}"]
      .get({
        params: { id: "1KCSPY1glIKqW2TotWuXOR" },
        headers: {
          Authorization: `Bearer ${import.meta.env.VITE_APP_SPOTIFY_TOKEN}}`,
        },
      })
      .then((res) => {
        if (res.ok) {
          return res.json();
        }

        throw new Error("Something went wrong");
      })
      .then((artist) => {
        setArtist(artist);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  if (!artist) {
    return <>Loading...</>;
  }

  return <>{artist.name}</>;
}

export default App;

```

It works without any problem, but I have the TS error on VSCode:
```js
// Type instantiation is excessively deep and possibly infinite.ts(2589)
```

Is there any solution to this?

**Comment by ardatan** [comment]
Could you reproduce it on CodeSandbox?

**Reply by p3yman** [reply]
No, but the TypeScript doesn't seem to work normally on CodeSandbox for me:
<img width="428" alt="Screenshot 2023-07-25 at 10 35 54" src="https://github.com/ardatan/feTS/assets/2673262/2076c33e-8b22-4483-a0a3-59d13d59879a">

https://codesandbox.io/p/sandbox/red-field-fs4zdm?file=%2Fsrc%2FApp.tsx%3A16%2C20

Either way, I also put it in a repo, if you want to have a closer look. Really appreciate it.
https://github.com/p3yman/fets-demo

**Reply by ardatan** [reply, marked as answer]
Could you try with the latest version?

**Reply by p3yman** [reply]
Upgraded all packages & restarted the TS server, but the same error is there.

**Reply by p3yman** [reply]
I pushed an update. It's fixed by extracting the fetch. Thanks for following this up
